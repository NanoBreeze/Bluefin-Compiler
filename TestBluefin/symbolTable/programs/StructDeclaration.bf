struct A {
	int a;
	int b;
	void f() {}
	int c;
} ;

struct B {
	int a = 9;
	int d = a;
} ;

struct C {
	A a;
	B b;
	int c;
	B bb = b;
};