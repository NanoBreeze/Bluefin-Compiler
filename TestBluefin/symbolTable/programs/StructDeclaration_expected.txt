declare - A - c_struct - t_A
setCurrentScope - Level 1
declare - a - c_var - t_int
declare - b - c_var - t_int
resolve - void - c_builtinType - t_void
declare - f - c_function - t_void
enterScope - Level 2
exitScope - Level 2
declare - c - c_var - t_int
exitScope - Level 1
declare - B - c_struct - t_B
setCurrentScope - Level 1
declare - a - c_var - t_int
declare - d - c_var - t_int
exitScope - Level 1
declare - C - c_struct - t_C
setCurrentScope - Level 1
declare - a - c_var - t_A
declare - b - c_var - t_B
declare - c - c_var - t_int
exitScope - Level 1
resolve - int - c_builtinType - t_int
resolve - int - c_builtinType - t_int
resolve - int - c_builtinType - t_int
resolve - int - c_builtinType - t_int
resolve - int - c_builtinType - t_int
resolve - A - c_struct - t_A
resolve - B - c_struct - t_B
resolve - int - c_builtinType - t_int
